hathor/p2p/sync_version.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2021 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from enum import Enum
+from functools import total_ordering
+
+
+@total_ordering
+class SyncVersion(Enum):
+    # XXX: These values are used in the protocol to negotiate commonly supported versions, changing it will cause peers
+    #      to no match different values and in turn not select a certain protocol, this can be done intentionally, for
+    #      example, peers using `v2-fake` (which just uses sync-v1) will not connect to peers using `v2-alpha`, and so
+    #      on.
+    V1 = 'v1'
+    V2 = 'v2-fake'  # uses sync-v1 to mock sync-v2
+
+    def __str__(self):
+        return f'sync-{self.value}'
+
+    def get_priority(self) -> int:
+        """Numerical indication used to sort preferred versions, higher values are preferred over lower values.
+
+        In practice this is used to sort versions.
+        """
+        # XXX: these values are only used internally and in memory, there is no need to keep them consistency, for
+        #      example, if we need more granularity, we can just add a 0 to all values and use the values in between,
+        #      although this shouldn't really be necessary
+        if self is SyncVersion.V1:
```

Comment:
Should we create a dict in the class?

Reply:
Makes sense. I think it might not be completely straightforward (because we can't access the class before creating it and accessing the members (`V1`, `V2`) will give their pre-metaclass values (`str`). But I think a dict right after the class would be great. Ideally in a few years when we can use [Python 3.10's match statements](https://docs.python.org/3.10/whatsnew/3.10.html#pep-634-structural-pattern-matching) this would be much cleaner IMO.

Reply:
I agree. This implementation seems good enough. Forget it. Thanks!